agent/base

```diff
@@ -14,17 +14,12 @@ if [[ -z "$pbench_run" ]]; then
     if [[ -z "$pbench_run" ]]; then
 	pbench_run=/var/lib/pbench-agent
     fi
+    export pbench_run
+else
     if [[ ! -d $pbench_run ]]; then
-        echo "[ERROR] pbench run directory, ${pbench_run}, does not exist" >&2
+        echo "[ERROR] the provided pbench run directory, ${pbench_run}, does not exist" >&2
```

Comment:
That does not create the directory: it exits with an error if it does not exist. So that contradicts what the commit message says, right?

Reply:
No, but since this is now in an else clause, we only consider the missing `$pbench_run` directory as an error if a caller of `base` set the environment variable to a directory that does not exist.

The only caller that does that is `unittests`.

Now, if `$pbench_run` is not provided as an environment variable, then we ensure by line 17 above that `$pbench_run` has a value, and line 28 below will create it.